docs/developer-docs/build/languages/motoko-index.md

```diff
@@ -0,0 +1,163 @@
+
+ The Motoko compiler, documentation and other tooling is [open source](https://github.com/dfinity/motoko) and released under the Apache 2.0 license. Contributions are welcome.
+
+ ## Native canister support
+
+ Motoko has native support for Internet Computer canisters.
+
+ A canister (or canister for short) is expressed as a Motoko actor. An actor is an autonomous object that fully encapsulates its state and communicates with other actors only through asynchronous messages.
+
+ ``` motoko
+ actor Counter {
+   var value = 0;
+   public func inc() : async Nat {
+     value += 1;
+     return value;
+   };
+ }
+ ```
+
+ ## Code sequentially in direct style
+
+ On the Internet Computer, canisters can communicate with other canisters by sending asynchronous messages.
+
+ Asynchronous programming is hard, so Motoko enables you to author asynchronous code in much simpler, sequential style. Asynchronous messages are function calls that return a *future*, and the `await` construct allows you to suspend execution until a future has completed. This simple feature avoids the "callback hell" of explicit asynchronous programming in other languages.
+
+ ``` motoko
+ actor Factorial {
+   var last = 1;
+   public func next() : async Nat {
+     last *= await Counter.inc();
+     return last;
+   }
+ };
+ ignore await Factorial.next();
+ ignore await Factorial.next();
+ await Factorial.next();
+ ```
+
+ ## Modern type system
+
+ Motoko has been designed to be intuitive to those familiar with JavaScript and other popular languages, but offers modern features such as sound structural types, generics, variant types, and statically checked pattern matching.
+
+ ``` motoko
+ type Tree<T> = {
+   #leaf : T;
+   #branch : {left : Tree<T>; right : Tree<T>};
+ };
+ func iterTree<T>(tree : Tree<T>, f : T -> ()) {
+   switch (tree) {
+     case (#leaf(x)) { f(x) };
+     case (#branch{left; right}) {
+       iterTree(left, f);
+       iterTree(right, f);
+     };
+   }
+ };
+ // Compute the sum of all leaf nodes in a tree
+ let tree = #branch { left = #leaf 1; right = #leaf 2 };
+ var sum = 0;
+ iterTree<Nat>(tree, func (leaf) { sum += leaf });
+ sum
+ ```
+
+ ## Autogenerated IDL files
+
+ A Motoko actor always presents a typed interface to its clients as a suite of named functions with argument and (future) result types.
+
+ The Motoko compiler (and SDK) can emit this interface in a language neutral format called Candid, so other canisters, browser resident code and smart phone apps that support Candid can use the actor�s services. The Motoko compiler can consume and produce Candid files, allowing Motoko to seamlessly interact with canisters implemented in other programming languages (provided they support Candid).
```

Comment:
Another weird char.

Reply:
Why does this file exist? It seems to replicate the content of motoko/doc/md/motoko.md from the motoko repo. Is it to avoid a clash with file `motoko` above?

Reply:
Ah. It got renamed and I missed it.